packages/vite/src/node/server/index.ts

```diff
@@ -499,6 +499,15 @@ export async function createServer(
     middlewares.use(corsMiddleware(typeof cors === 'boolean' ? {} : cors))
   }
 
+  // escape colon in path
+  middlewares.use(function escapeColonMiddleware(req, res, next) {
+    if (req.url) {
+      // example url: /node_modules/.vite/node:stream.js?v=xxxxxxxx
+      req.url = req.url.replace(/:/g, '_')
```

Comment:
This regex doesn't look safe at all :eyes: 
Is it really wanted to find all colons? Also that one in query params?

Beside that, I think this would need a test

Reply:
if we want to do this, shouldn't it only happen for `node:`?